src/nft/components/details/AssetPriceDetails.tsx

```diff
@@ -1,3 +1,205 @@
+              <ThemedText.BodySecondary lineHeight={'24px'}>
+                {ethNumberStandardFormatter(asset.priceInfo.USDPrice, true, true)}
+              </ThemedText.BodySecondary>
+            </>
+          ) : (
+            <ThemedText.BodySecondary fontSize="14px" lineHeight={'20px'}>
+              Get the best price for your NFT by selling with Uniswap.
+            </ThemedText.BodySecondary>
+          )}
+        </PriceRow>
+        {expirationDate && (
+          <ThemedText.BodySecondary fontSize={'14px'}>Sale ends: {timeLeft(expirationDate)}</ThemedText.BodySecondary>
+        )}
+        {!listing ? (
+          <BuyNowButton assetInBag={false} margin={true} useAccentColor={true} onClick={() => navigate('/profile')}>
+            <ThemedText.SubHeader lineHeight={'20px'}>List</ThemedText.SubHeader>
+          </BuyNowButton>
+        ) : (
+          <>
+            <BuyNowButton assetInBag={false} margin={true} useAccentColor={false} onClick={() => navigate('/profile')}>
+              <ThemedText.SubHeader lineHeight={'20px'}>Adjust listing</ThemedText.SubHeader>
+            </BuyNowButton>
+            <BuyNowButton assetInBag={true} margin={false} useAccentColor={false} onClick={() => navigate('/profile')}>
+              <ThemedText.SubHeader lineHeight={'20px'}>Cancel listing</ThemedText.SubHeader>
+            </BuyNowButton>
+          </>
+        )}
+      </BestPriceContainer>
+    </Container>
+  )
+}
+
+export const NotForSale = ({ collection }: { collection: CollectionInfoForAsset }) => {
+  const theme = useTheme()
+
+  return (
+    <BestPriceContainer>
+      <NotForSaleContainer>
+        <CancelListingIcon width="79px" height="79px" color={theme.textTertiary} />
+        <ThemedText.SubHeader fontWeight={500} lineHeight="24px">
+          Not for sale
+        </ThemedText.SubHeader>
+        <DiscoveryContainer>
+          <ThemedText.BodySecondary fontSize="14px" lineHeight="20px">
+            Discover similar NFTs for sale in
+          </ThemedText.BodySecondary>
+          <ThemedText.Link lineHeight="20px">{collection.collectionName}</ThemedText.Link>
+        </DiscoveryContainer>
+      </NotForSaleContainer>
+    </BestPriceContainer>
+  )
+}
+
+export const AssetPriceDetails = ({ asset, collection }: AssetPriceDetailsProps) => {
+  const { account } = useWeb3React()
+  const cheapestOrder = asset.sellorders && asset.sellorders.length > 0 ? asset.sellorders[0] : undefined
+  const expirationDate = cheapestOrder ? new Date(cheapestOrder.orderClosingDate) : undefined
+  const itemsInBag = useBag((s) => s.itemsInBag)
+  const addAssetToBag = useBag((s) => s.addAssetToBag)
+  const removeAssetFromBag = useBag((s) => s.removeAssetFromBag)
+
+  const assetInBag = useMemo(() => {
+    return itemsInBag.some((item) => item.asset.tokenId === asset.tokenId && item.asset.address === asset.address)
+  }, [itemsInBag, asset])
+
+  const isOwner = account?.toLowerCase() === asset.owner.toLowerCase()
```

Comment:
Yeah this is actually a backend problem the backend is returning the wrong type here because it hasn't refreshed the metadata of that token. It shouldn't change just because the backend is using old data.